### ISSUE TITLE
Default appender should log using `print` if `console.log` is not available

### ISSUE THREAD

**Issue by polymeris** [ptaoussanis/timbre#220]
Would allow use with [Rhino](https://developer.mozilla.org/en-US/docs/Mozilla/Projects/Rhino/Shell#print.28.5Bexpr_....5D.29).

**Comment by ptaoussanis** [comment]
Hi Camilo,

Just to clarify: is there a reason you can't / would prefer not to enable the included :println appender in this case?

**Comment by polymeris** [comment]
I can live with that, just thought it would be more consistent if both CLJS and CLJ behaved the same by default (printing to stdout).

**Comment by ptaoussanis** [comment]
>  just thought it would be more consistent if both CLJS and CLJ behaved the same by default (printing to stdout)

That makes sense, and it's a good goal. In practice, the trouble is that the concept of "stdout" isn't quite consistent between CLJS and CLJ. In particular, CLJS can operate in different environments (e.g. without a console) and doesn't provide a default implementation for println (specifically, doesn't offer a default implementation for `*print-fn*`). Moreover it's currently difficult to detect if a "real" `*print-fn*` is actually in place or not. Without one, the default behaviour is for `println` calls to throw.

tl;dr The platform differences are significant enough here to make it so that you do actually need to be clear about what you want (even though it's less convenient).

Hope that makes sense?

**Comment by polymeris** [comment]
> Hope that makes sense?

Yes, absolutely. Thanks for the thorough explanation!
